libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -546,12 +561,23 @@ void AP_Baro::init(void)
 #else
         ADD_BACKEND(AP_Baro_MS56XX::probe(*this,
                                           std::move(hal.i2c_mgr->get_device(_ext_bus, HAL_BARO_MS5611_I2C_ADDR))));
+
+ #if CONFIG_HAL_BOARD == HAL_BOARD_F4LIGHT // we don't know which baro user will solder
+
+        ADD_BACKEND(AP_Baro_BMP280::probe(*this,
+                                          std::move(hal.i2c_mgr->get_device(_ext_bus, HAL_BARO_BMP280_I2C_ADDR))));
+        ADD_BACKEND(AP_Baro_BMP085::probe(*this,
+                                          std::move(hal.i2c_mgr->get_device(_ext_bus, HAL_BARO_BMP085_I2C_ADDR))));
+ #endif                                         
 #endif
     }
 
+#if CONFIG_HAL_BOARD != HAL_BOARD_F4LIGHT // most boards requires external baro
```

Comment:
I have no comment on removing the need for a baro, but the comment is wrong. The code below doesn't require an external baro nor any board that I know of requires it; it simply requires that at least one baro is connected.

Reply:
User should can to use board without any baro. Arming check will prevent him from flight anyway

Reply:
it would be nice to support boards with no baro in limited flight modes (eg. ACRO mode in copter). It would probably be quite an intrusive change though.

Reply:
I thought about it and tried to do it, but it took too long